src/vscodeTelemetry.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import TelemetryReporter from './telemetryReporter';
+
+/**
+ * Intruments telemetry for VS Code Extension API triggering or triggered by user actions.
+ */
+export function instrumentVSCodeTelemetry(telemetry: TelemetryReporter): void {
```

Comment:
Thinking more about this, this looks too hacky :smile: , how about having a `notificationService` class instead that just wraps the vscode api, adds the telemetry stuff and use that instead in all places

Reply:
let's do it